### PR TITLE
OOP Library : 1 - Add person, student and teacher classes

### PR DESCRIPTION
**Pull request by lmunozm1702**
### Deliver the 1st. part:

- Create classes as required: Person, Teacher and Student.

image

- Each class should be saved in a separate file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,36 +60,32 @@ After you're finished please remove all the comments and instructions!
 
 <!-- PROJECT DESCRIPTION -->
 
-# 📖 [your_project_name] <a name="about-project"></a>
+# 📖 [School Library] <a name="about-project"></a>
 
 > Describe your project in 1 or 2 sentences.
 
-**[your_project__name]** is a...
+**[school library]** is a OOP and Ruby based application to ecord what books are in the library and who borrows them.
```
Comment (danmainah) [inline README.md]: - [x] [optional] It would have been nice if you corrected spelling error in the name record

Comment (danmainah) [reply README.md]: - [x] kindly remove the commented section of the readme file, its not good practice to have commented code in the codebase

File: persons.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+class Person
+  # creating accessors
+  attr_accessor :name, :age, :id
```
Comment (danmainah) [inline persons.rb]: - [x] @lmunozm1702 kindly ensure only name and age have setter and getter, the id should only be a getter as in the project description
image

File: persons.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  def initialize(name, age = 'unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
```
Comment (danmainah) [inline persons.rb:10]: - [x] kindly add name as an instance variable, you can use `Random.rand(Random.rand(id numbers range params eg 1-1000)` to generate unique ids

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @lmunozm1702 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @lmunozm1702👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**